shell/platform/embedder/embedder.cc

```diff
@@ -712,6 +712,25 @@ inline flutter::PointerData::SignalKind ToPointerDataSignalKind(
   return flutter::PointerData::SignalKind::kNone;
 }
 
+// Returns the buttons for PointerData for the given buttons and change from a
+// FlutterPointerEvent.
+inline int64_t ToPointerDataButtons(
+    int64_t buttons,
+    flutter::PointerData::Change change) {
+  // `buttons` should be non-zero when and only when the pointer is down.
+  if (change == flutter::PointerData::Change::kDown ||
```

Comment:
I would recommend a `switch` rather than an `if` here so that if a new change type is added, the compiler with warn about the need to evaluate this code to decide how to handle it (since it may be down).

Reply:
Do you mean not using `default` but writing all options instead?

Reply:
Yes.